encode/src/main/scala/org/broadinstitute/monster/etl/encode/extract/EncodeExtractions.scala

```diff
@@ -0,0 +1,93 @@
+package org.broadinstitute.monster.etl.encode.extract
+
+import io.circe.JsonObject
+import org.broadinstitute.monster.etl.encode.extract.client.EncodeClient
+import cats.implicits._
+import org.apache.beam.sdk.transforms.GroupIntoBatches
+import org.apache.beam.sdk.values.KV
+import com.spotify.scio.values.SCollection
+import scala.collection.JavaConverters._
+
+class EncodeExtractions(client: EncodeClient) {
+
+  def extractEntities(
+    entryName: String,
+    entityType: String
+  ): SCollection[List[(String, String)]] => SCollection[JsonObject] =
+    _.transform(s"extractEntities - $entryName") {
+      _.flatMap { params =>
+        //generic operation on SCollection[List[(String, String)]] for all steps of extractions
+        client
+          .search(
+            entityType,
+            List("frame" -> "object", "status" -> "released") ::: params
+          )
+          .compile
+          .toList
+          .unsafeRunSync()
+      }
+    }
+
+  def getExperimentSearchParams(
+    entryName: String
+  ): SCollection[String] => SCollection[List[(String, String)]] =
+    _.transform(s"getExperimentSearchParams - $entryName") {
+      for {
+        assayType <- _
+      } yield {
+        List("assay_title" -> assayType)
+      }
+    }
+
+  def extractExperimentSearchParams(
+    entryName: String,
+    entityType: String
+  ): SCollection[List[(String, String)]] => SCollection[JsonObject] =
+    _.transform(s"extractExperimentSearchParams - $entryName") { collections =>
+      extractEntities(entityType, entityType)(collections)
+    }
+
+  def getIDParams(
+    entryName: String,
+    referenceField: String,
+    manyReferences: Boolean
+  ): SCollection[JsonObject] => SCollection[String] =
+    _.transform(s"getIDParams - $entryName") {
+      _.map {
+        _(referenceField)
```

Comment:
I'm probably to blame by example, but I think this is getting too `_`-heavy. I'd give a name to this object for clarity:
```scala
_.map(json => json(referenceField))
```